app/javascript/src/Settings/authentication-widget.jsx

```diff
@@ -37,6 +37,24 @@ const setValue = (el, val) => {
 }
 const setInputValue = (val) => (isReadOnly) => isReadOnly ? setting => setValue(setting, val) : setting => setting
 
+const clearPublicURLs = () => {
+  document.querySelector('#service_proxy_attributes_sandbox_endpoint').value = ''
+  document.querySelector('#service_proxy_attributes_endpoint').value = ''
+}
+
+const confirmDeploymentOptionChange = (event) => {
+  const isHostedToSelfManaged = event.target.value === 'self_managed'
+  const message = `This action will have the effect to ${isHostedToSelfManaged
+      ? 'clear out your Public Base URLs'
+      : 'set your Public Base URLs to a system generated'} as well as to ${ isHostedToSelfManaged
+      ? 'delete the corresponding routes in OpenShift'
+      : 'create the corresponding routes in OpenShift'}`
```

Comment:
It's hard to read like that. Has this been provided by docs?
```suggestion
  const message = isHostedToSelfManaged
    ? 'This action will have the effect to clear out your Public Base URLs as well as to delete the corresponding routes in Openshift'
    : 'This action will have the effect to set your Public Base URLs to a system generated as well as to create the corresponding routes in Openshift'
```